pkg/test/framework/components/echo/annotations.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package echo
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type AnnotationType string
+
+const (
+	ServiceAnnotation  AnnotationType = "service"
+	WorkloadAnnotation AnnotationType = "workload"
+)
+
+type Annotation struct {
+	Name    string
+	Type    AnnotationType
+	Default AnnotationValue
+}
+
+var (
+	// TODO: Keep this list up-to-date with pilot/pkg/kube/inject/inject.go
```

Comment:
just curious, how do you ensure this is the complete list of the annotations we support today?

Reply:
I don't. But as a test author, you can update this list.